feat!: Implementing OS-specific download buttons

plvzfq-rit opened the pull request:
<!--
☝️ PR title should follow conventional commits (https://conventionalcommits.org).
In particular, the title should start with one of the following types:

- docs: 📖 Documentation (updates to the documentation or readme)
- fix: 🐞 Bug fix (a non-breaking change that fixes an issue)
- feat: ✨ New feature/enhancement (a non-breaking change that adds functionality or improves existing one)
- feat!/fix!: ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- chore: 🧹 Chore (updates to the build process or auxiliary tools and libraries)
-->

### 🔗 Linked issue

<!-- If it resolves an open issue, please link the issue here. For example "Resolves #123" -->
Partially fixes #2176 
Fixes #2586

### 📚 Description

<!-- Describe your changes in detail -->
This pull request changes the download interface of the website. To detail, the old `Download JabRef` button is now changes based on the OS of the system (e.g., `Windows`, `Mac`, `Linux`). Windows should have buttons for `.msi` and `.zip`. For Linux, `.deb`, `.rpm`, and `.tar.gz`. For Mac (both `arm64` and `x86_64`), `.dmg` and `.pkg`. Consequently, changes in the `/download/[[os]]` path was necessary to implement redirection (expansion from just `['win', 'mac', 'linux']`

<!-- Why is this change required? What problem does it solve? -->
This pull request solves a user problem in which the web site automatically downloads a file which might not be apt for their package manager and or system architecture.

#### Original Implementation for Windows
![image](https://github.com/user-attachments/assets/0ec1638d-8194-4911-956a-0a0e71b3cf4c)

#### New Implementation for Windows
![image](https://github.com/user-attachments/assets/e3f83769-c731-429c-b7cd-326b74ccbca2)

#### New Implementation for Mac
![image](https://github.com/user-attachments/assets/7750f0d4-56c1-435f-bb93-95f4faf9d66e)

#### New Implementation for Linux
![image](https://github.com/user-attachments/assets/709e4ed7-a926-487f-b297-7b6c93436ccd)

Siedlerchr commented:
Sorry for the delay, looks good from my side!°

koppor commented:
@tobiasdiez What is the status here? This PR seems to be a good preparation for a download-button for JabRef 6.0-alpha.

koppor commented:
@plvzfq-rit I couldn't test. Can one download a macOS version on Windows? I didn't see an overview page on the screenshots.

Siedlerchr commented on pages/download/[[os]].vue:
```diff
@@ -22,15 +22,21 @@ definePageMeta({
     const os = to.params.os as string | undefined
-    if (os && ['win', 'mac', 'linux'].includes(os)) {
+    if (os && ['win_msi', 'win_zip', 'mac_arm64_dmg', 'mac_arm64_pkg', 'mac_x86_64_dmg', 'mac_x86_64_pkg', 'linux_deb', 'linux_rpm', 'linux_tar_gz'].includes(os)) {
       const { data } = await useFetch('/api/getLatestRelease')
```
We are in the process of phasing out fosshub, can you change it to github release?

plvzfq-rit replied:
I'll see what I could do+

plvzfq-rit replied:
Hi @Siedlerchr, just finished migrating the links from Fosshub to GitHub Release. There have been troubles on my end though with the ```useFetch``` function for some reason that I'm not sure of; this hinders me from fully checking if the implementation's good or not. Using it at the moment in my branch yields an undefined value. I checked the response attained by the function and it said that I had bad credentials. Investigating further led me to see that I do not have a ```GITHUB_REPO_TOKEN``` attribute in my .env file. Should I perhaps put a personal GitHub token in the .env file for testing purposes? 

For now though, I'll try to work on the overview page. Much thanks!

plvzfq-rit commented:
@koppor, unfortunately, as far as I have implemented it, there would not be a way for a Windows user to download a macOS package, aside from downloading it directly from the fosshub page upon redirect (will be changing it to the GitHub in the coming days). I was planning on adding that functionality more explicitly in a later pull request (creating an overview page / a page listing download links for the latest version of each OS/architecture), as I had discussed with @tobiasdiez . Should I perhaps include it within this pull request or continue to have it be in a different one?

plvzfq-rit commented:
Just found a bug which affected the href's in the buttons. Should be fixed now though. Apologies for the inconvenience;;;

plvzfq-rit commented:
By the by, should we still work on an overview page? I think linking to the [GitHub releases](https://github.com/JabRef/jabref/releases/latest) page might already be good for that. Less code to maintain as well.

Siedlerchr commented:
Happy new Year, I think using github releases as overview page is totally fine^

Siedlerchr commented:
There is one lint error:
>  /home/runner/work/JabRefOnline/JabRefOnline/components/LandingPageDownload.client.vue
Error:   92:7  error    'downloadUrl' is assigned a value but never used                                        @typescript-eslint/no-unused-vars
Warning:   92:7  warning  'downloadUrl' is assigned a value but never used. Allowed unused vars must match /^_/u  unused-imports/no-unused-vars

ThiloteE commented:
I am missing a pointer to a "portable version". Those are the `.zip` and `.tar.gz` files, I suppose? I don't know Mac, so I don't know what it would be there. Would be nice, if we somehow could have a short description for Users that are not knowledgable about file endings.

plvzfq-rit commented:
Hi @Siedlerchr, @ThiloteE, I've put in the GitHub releases page as a link to the overiew page instead of the fosshub one. Also changed the associated test and changed the text on the .zip and .tar.gz button to "Windows Portable" and "Linux Portable," respectively. Also ran Prettier and removed unused imports. Let me know if there's anything else+

koppor commented:
@plvzfq-rit Thank you for keeping working on this! 💪

Think, we should merge this.
